core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -1510,6 +1511,9 @@ private boolean rerunWF(String workflowId, String taskId, Map<String, Object> ta
             workflow.getTasks().forEach(task -> executionDAOFacade.removeTask(task.getTaskId()));
             // Set workflow as RUNNING
             workflow.setStatus(WorkflowStatus.RUNNING);
+            // Reset failure reason from previous run to default
+            workflow.setReasonForIncompletion(null);
+            workflow.setFailedReferenceTaskNames(new HashSet<>());
```

Comment:
I think you would need to make similar changes in line 1585 below.

Reply:
Did the changes in this [commit](https://github.com/Netflix/conductor/pull/1822/commits/9b9b11d2b3d8e845be4891bf28ce256ea7f800b0)
is it correct?